ui/src/components/btn/QasBtn.vue

```diff
@@ -1,10 +1,18 @@
 <template>
   <q-btn ref="button" class="qas-btn" v-bind="attributes">
-    <slot />
-
     <template v-for="(_, name) in nonDefaultSlots" #[name]="context">
       <slot :name="name" v-bind="context || {}" />
     </template>
+
+    <span class="items-center justify-center no-wrap row text-center">
+      <q-spinner v-if="hasLeftSpinner" :class="iconClasses" size="sm" />
+      <q-icon v-if="hasIcon" :class="iconClasses" :name="props.icon" size="xs" />
+      <span v-if="showLabel">{{ props.label }}</span>
+      <q-spinner v-if="hasRightSpinner" :class="iconRightClasses" size="sm" />
+      <q-icon v-if="hasIconRight" :class="iconRightClasses" :name="props.iconRight" size="xs" />
```

Comment:
n daria pra renderizar dinamicamente sem a necessidade desses monte de v-if? talvez com component is?

Reply:
não pq não é um v-if e v-else, e sim cada um tem suas validações pra serem exibidos.